### PR TITLE
fix build with 'USE_HDFS' on windows

### PR DESCRIPTION
**Pull request by zhli1142015**
Summary:
Build with "USE_HDFS" failed with below errors on Windows. This PR is trying to fix them
Severity	Code	Description	Project	File	Line	Suppression State
Error (active)	E0020	identifier "ssize_t" is undefined	rocksdb	D:\Git\rocksdb\rocksdb\env\env_hdfs.cc	127	
Error (active)	E1696	cannot open source file "sys/time.h"	rocksdb	D:\Git\rocksdb\rocksdb\env\env_hdfs.cc	15	
Error	C2065	'pthread_t': undeclared identifier	rocksdb	d:\git\rocksdb\rocksdb\hdfs\env_hdfs.h	166	
Error	C3861	'pthread_self': identifier not found	rocksdb	d:\git\rocksdb\rocksdb\hdfs\env_hdfs.h	167	
Error	C1083	Cannot open include file: 'sys/time.h': No such file or directory	rocksdb	d:\git\rocksdb\rocksdb\env\env_hdfs.cc	15	
Error	C2065	'pthread_t': undeclared identifier	db_bench	d:\git\rocksdb\rocksdb\hdfs\env_hdfs.h	166	
Error	C3861	'pthread_self': identifier not found	db_bench	d:\git\rocksdb\rocksdb\hdfs\env_hdfs.h	167	

Test plan:
1. manually test build with "USE_HDFS" on Windows, verified HDFS Env related function by db_bench.exe.
D:\Git\rocksdb\build\Debug>db_bench.exe --hdfs="abfs://test@rdbtest2.dfs.core.windows.net" --num=100 --benchmarks="fillseq,readseq,fillseekseq" --db="abfs://test@rdbtest2.dfs.core.windows.net/test"
2020-06-05 20:42:21,102 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
2020-06-05 20:42:22,646 WARN utils.SSLSocketFactoryEx: Failed to load OpenSSL. Falling back to the JSSE default.
Initializing RocksDB Options from the specified file
Initializing RocksDB Options from command-line flags
RocksDB:    version 6.10
Keys:       16 bytes each
Values:     100 bytes each (50 bytes after compression)
Entries:    100
Prefix:    0 bytes
Keys per prefix:    0
RawSize:    0.0 MB (estimated)
FileSize:   0.0 MB (estimated)
Write rate: 0 bytes/second
Read rate: 0 ops/second
Compression: Snappy
Compression sampling rate: 0
Memtablerep: skip_list
Perf Level: 1
WARNING: Assertions are enabled; benchmarks unnecessarily slow
------------------------------------------------
Initializing RocksDB Options from the specified file
Initializing RocksDB Options from command-line flags
DB path: [abfs://test@rdbtest2.dfs.core.windows.net/test]
fillseq      :    1138.350 micros/op 877 ops/sec;    0.1 MB/s
DB path: [abfs://test@rdbtest2.dfs.core.windows.net/test]
readseq      :      63.580 micros/op 15627 ops/sec;    1.7 MB/s
DB path: [abfs://test@rdbtest2.dfs.core.windows.net/test]
fillseekseq  :      45.615 micros/op 21762 ops/sec;

### CODE REVIEW

**Comment by zhli1142015** [comment]
@yhchiang , @ajkr , could you please help review?

File: hdfs/env_hdfs.h [hunk 1]
```diff
@@ -162,10 +162,7 @@ class HdfsEnv : public Env {
     return posixEnv->TimeToString(number);
   }
 
-  static uint64_t gettid() {
-    assert(sizeof(pthread_t) <= sizeof(uint64_t));
-    return (uint64_t)pthread_self();
-  }
+  static uint64_t gettid() { return Env::Default()->GetThreadID(); }
```
Comment (riversand963) [inline hdfs/env_hdfs.h:165]: Nit:
`posixEnv->GetThreadID()`

Comment (zhli1142015) [reply hdfs/env_hdfs.h:165]: `gettid()` is static method, thanks.

File: env/env_hdfs.cc [hunk 2]
```diff
@@ -124,8 +123,8 @@ class HdfsReadableFile : virtual public SequentialFile,
     Status s;
     ROCKS_LOG_DEBUG(mylog, "[hdfs] HdfsReadableFile preading %s\n",
                     filename_.c_str());
-    ssize_t bytes_read = hdfsPread(fileSys_, hfile_, offset,
-                                   (void*)scratch, (tSize)n);
+    tSize bytes_read =
+        hdfsPread(fileSys_, hfile_, offset, (void*)scratch, (tSize)n);
```
Comment (riversand963) [inline env/env_hdfs.cc:127]: Sorry for the last minute comment.
Here, static_cast is preferred.

File: hdfs/env_hdfs.h [hunk 3]
```diff
@@ -210,7 +207,7 @@ class HdfsEnv : public Env {
                            remaining.substr(0, rem));
 
     tPort port;
-    port = atoi(portStr.c_str());
+    port = static_cast<tPort>(atoi(portStr.c_str()));
```
Comment (riversand963) [inline hdfs/env_hdfs.h:210]: Nit:
```
tPort port = static_cast<tPort>(atoi(portStr.c_str()));
```

Comment (riversand963) [reply hdfs/env_hdfs.h:210]: Also, can an illegal value of `portStr` lead to unhandled error?

Comment (riversand963) [reply hdfs/env_hdfs.h:210]: I guess we can leave this as is.

File: env/env_hdfs.cc (hunk 2 above)
Comment (zhli1142015) [reply env/env_hdfs.cc:127]: i prefer `tSize` here, return value type from `hdfsPread`, thanks.

Comment (riversand963) [reply env/env_hdfs.cc:127]: I mean something like the following.
```
tSize bytes_read = hdfsPread(fileSys_, hfile_, offset, static_cast<void*>(scratch), static_cast<tSize>(n));
```

File: hdfs/env_hdfs.h (hunk 3 above)
Comment (zhli1142015) [reply hdfs/env_hdfs.h]: thanks, updated.

File: env/env_hdfs.cc (hunk 2 above)
Comment (zhli1142015) [reply env/env_hdfs.cc]: get it, thanks, updated.

**Pull request closed without merging**